### DISCUSSION TITLE
Table with multiple statuses

### DISCUSSION THREAD

**Discussion by char-mainewang** [twilio-labs/paste#2591] [Q&A] [answered]
hey team, i'm designing a table that contains 2 statuses:
- one to indicate if it's on or off
- another to indicate the status of the job (in progress, success, failed). 

we currently use the evergreen's status component  pretty consistently across app to indicate on/off or if something is live.
<img width="425" alt="image" src="https://user-images.githubusercontent.com/87334135/182981562-8aa1f015-a0e2-45eb-98f3-bf949c3366f2.png">

I'm considering using a badge to indicate the job status to add visual contrast and draw attention to the job status as that should take precedent over the on/off status in most cases. here's a link to [the table design](https://www.figma.com/file/WgVbZObTywZc44hX5RzxR9/%5BDesign%5D-Reverse-ETL---Post-design-sprint?node-id=5030%3A220534). would love to know if I should stick to using the status component or use the badge, any guidance from the team on what to do in this situation would be greatly appreciated :)

**Comment by loreina** [comment]
Hi @char-mainewang, I brought this back to the team and we'll get back to you soon. Thanks for being patient!

**Comment by loreina** [comment, marked as answer]
Hi @char-mainewang, we have a ticket in our backlog to add Badge to the [Status pattern](https://paste.twilio.design/patterns/status) since we built Badge after the pattern was first published. 

For your use case, we would recommend to either use:
1. [Process status](https://paste.twilio.design/patterns/status#process-status) for "Run Status" and [connectivity status](https://paste.twilio.design/patterns/status#connectivity-status) for "Mapping Status"
2. [Badge](https://paste.twilio.design/components/badge) for "Run Status" and [connectivity status](https://paste.twilio.design/patterns/status#connectivity-status) for "Mapping Status"

I made a couple mockups of the 2 recommendations here: https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Paste-Feedback?node-id=1708%3A377967. You can find the [Status pattern component](https://www.figma.com/file/S4z0Kqjb9AYosnkRQldLte/Paste-Patterns?node-id=308%3A27277) in the Paste Patterns library, and [Badge](https://www.figma.com/file/E6KUvMhioUmAgN0nwmReTM/Paste-Components?node-id=4379%3A41347) in the Paste Components library.

Let us know what you decide to go with (especially if you go with Badge) and we can add your designs as a use case for when we get around to adding Badge to the Status pattern!

**Reply by char-mainewang** [reply]
Hi, just realized i never updated this github post, but I decided to use badges in my original evergreen designs. 

As we're transitioning to Paste, i'm curious to get y'all thoughts on whether we should use badges to represent sync status (for different things syncing) consistently across the app even if it's not displayed with a connectivity status or only use badge when it's displayed with a connectivity status?

**Reply by SiTaggart** [reply]
Wow, this is an old thread! We now have a [Status Badge component](https://paste.twilio.design/components/status-badge), it comes with process and connectivity variants. The Process Status variants might work for you.

As for consistent representation, that's up to you and the app, I guess. There is some concern on the team that badges in tables can become pretty noisy visually, but ultimately I think that's your trade-off call to make. Habit-forming consistency vs Ideal visual treatments.

**Reply by charleneyjh1** [reply]
Hi @SiTaggart 

Is the status badge component meant to replace status pattern? I'm not sure when should i use the status with border and when should i use the one without border. Could you provide more clarity there? Planning to use them in a table to indicate whether a destination is live or disabled, or has an error. 

<img width="108" alt="Screenshot 2023-10-20 at 12 59 52 PM" src="https://github.com/twilio-labs/paste/assets/56697620/f5dab1ed-78e2-4901-912a-d9895eed8f5d">
<img width="98" alt="Screenshot 2023-10-20 at 12 59 58 PM" src="https://github.com/twilio-labs/paste/assets/56697620/7a1ee339-2d4e-44d9-92e7-79e0b745a7e1">

**Reply by serifluous** [reply]
Hey @charleneyjh1 ! We unfortunately still need to do some work in updating the Status pattern.

Generally, you'd show status without the border (just icon + text) in a table, and then use the Status Badge outside of a table: https://paste.twilio.design/patterns/status#status-badges

(Some of the examples on the Status pattern page don't reflect that guidance right now.)

**Reply by charleneyjh1** [reply]
I see, thanks for clarifying! that helps a lot.
